update_unix.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/user"
+	"path"
+	"path/filepath"
+)
+
+// RunUpdate runs the update on the current tgf executable
+func RunUpdate() bool {
+	cloneExecutableAt(filepath.Join(getTgfHomeDirectory(), "tgf"))
+
+	os.Setenv("TGF_PATH", getTgfHomeDirectory())
+
+	updateScript, err := fetchUpdateScript()
+	if err != nil {
+		printWarning("Error fetching update script: ", err)
+	}
+
+	output, err := exec.Command("bash", "-c", updateScript).CombinedOutput()
+	if err != nil {
+		printWarning("Error running update script: ", err)
+		Println(string(output))
+	}
+
+	return err == nil
+}
+
+func fetchUpdateScript() (string, error) {
+	resp, err := http.Get("https://raw.githubusercontent.com/coveo/tgf/master/get-latest-tgf.sh")
+	if err != nil {
+		printWarning("Error fetching update script", err)
+	}
+
+	textResponse, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		printWarning("Error reading request body", err)
+	}
+
+	updateScript := string(textResponse)
+
+	return updateScript, err
+}
+
+// ReRunCopy calls tgf with the provided arguments on Unix
+func ReRunCopy() {
+	homeExecutablePath := getTgfHomeExecutable()
+	currentExecutablePath := getCurrentExecutablePath()
+
+	newArgs := append(os.Args[1:], "--copy-executable", currentExecutablePath)
+
+	cmd := exec.Command(homeExecutablePath, newArgs...)
+	cmd.Stdout = os.Stdout
+	cmd.Stdin = os.Stdin
+	cmd.Stderr = os.Stderr
+	cmd.Run()
+	cmd.Process.Release()
```

Comment:
What's the use for this?

Reply:
Since the **ReRun** calls the `copy` executable, we have to detach the subprocess. Otherwise, we end up with `busy file` error when trying to replace the `original` executable. 
Since the parent process waits for the child's execution by default, `Release()` makes it run separately.

Reply:
Never mind, it's not necessary.